cli/parse_args.ts

```diff
@@ -450,316 +473,250 @@ export function parseArgs<
-        aliases[key] = val as Array<string>;
-      }
-      const aliasesForKey = getForce(aliases, key);
-      for (const alias of aliasesForKey) {
-        aliases[alias] = [key].concat(aliasesForKey.filter((y) => alias !== y));
-      }
+      const val = (alias as Record<string, unknown>)[key];
+      assert(val !== undefined);
+      const aliases = Array.isArray(val) ? val : [val];
+      aliasMap.set(key, new Set(aliases));
+      const set = new Set([key, ...aliases]);
+      aliases.forEach((alias) => aliasMap.set(alias, set));
     }
   }
 
-  if (boolean !== undefined) {
+  if (boolean) {
     if (typeof boolean === "boolean") {
-      flags.allBools = !!boolean;
+      allBools = boolean;
     } else {
-      const booleanArgs: ReadonlyArray<string> = typeof boolean === "string"
-        ? [boolean]
-        : boolean;
-
+      const booleanArgs = Array.isArray(boolean) ? boolean : [boolean];
       for (const key of booleanArgs.filter(Boolean)) {
-        flags.bools[key] = true;
-        const alias = get(aliases, key);
-        if (alias) {
-          for (const al of alias) {
-            flags.bools[al] = true;
-          }
-        }
+        booleanSet.add(key);
+        aliasMap.get(key)?.forEach((al) => {
+          booleanSet.add(al);
+        });
       }
     }
   }
 
-  if (string !== undefined) {
-    const stringArgs: ReadonlyArray<string> = typeof string === "string"
-      ? [string]
-      : string;
-
+  if (string) {
+    const stringArgs = Array.isArray(string) ? string : [string];
     for (const key of stringArgs.filter(Boolean)) {
-      flags.strings[key] = true;
-      const alias = get(aliases, key);
-      if (alias) {
-        for (const al of alias) {
-          flags.strings[al] = true;
-        }
-      }
+      stringSet.add(key);
+      aliasMap.get(key)?.forEach((al) => stringSet.add(al));
     }
   }
 
-  if (collect !== undefined) {
-    const collectArgs: ReadonlyArray<string> = typeof collect === "string"
-      ? [collect]
-      : collect;
-
+  if (collect) {
+    const collectArgs = Array.isArray(collect) ? collect : [collect];
     for (const key of collectArgs.filter(Boolean)) {
-      flags.collect[key] = true;
-      const alias = get(aliases, key);
-      if (alias) {
-        for (const al of alias) {
-          flags.collect[al] = true;
-        }
-      }
+      collectSet.add(key);
+      aliasMap.get(key)?.forEach((al) => collectSet.add(al));
     }
   }
 
-  if (negatable !== undefined) {
-    const negatableArgs: ReadonlyArray<string> = typeof negatable === "string"
-      ? [negatable]
-      : negatable;
-
+  if (negatable) {
+    const negatableArgs = Array.isArray(negatable) ? negatable : [negatable];
     for (const key of negatableArgs.filter(Boolean)) {
-      flags.negatable[key] = true;
-      const alias = get(aliases, key);
-      if (alias) {
-        for (const al of alias) {
-          flags.negatable[al] = true;
-        }
-      }
+      negatableSet.add(key);
```

Comment:
This does the same as the removed code, but more readable with the use of `Set`.